RecoParticleFlow/PFProducer/plugins/PFPhotonTranslator.cc

```diff
@@ -962,8 +962,9 @@ void PFPhotonTranslator::createPhotons(reco::VertexCollection &vertexCollection,
     showerShape.maxEnergyXtal = egPhotonRef_[iphot]->maxEnergyXtal();
     showerShape.sigmaEtaEta = egPhotonRef_[iphot]->sigmaEtaEta();
     showerShape.sigmaIetaIeta = egPhotonRef_[iphot]->sigmaIetaIeta();
-    showerShape.hcalDepth1OverEcal = egPhotonRef_[iphot]->hadronicDepth1OverEm();
-    showerShape.hcalDepth2OverEcal = egPhotonRef_[iphot]->hadronicDepth2OverEm();
+    for (int id = 0; id < 7; ++id) {
```

Comment:
`7` is hardcoded multiple times here. Can you make it configurable?

Reply:
I don't see why should this be configurable, how about looping up to showerShape.hcalOverEcal.size() instead?

See also my reply to your comment about (7, 4) below.

Reply:
if someone 10y later needs to change 7->8 for whatever reason, it will be very difficult to trace down all the places across all the different files. moreover, if one place is missed, there is no compile time error, and the code will still run but possibly produce incorrect results.